src/main/java/org/generationcp/middleware/service/impl/study/StudyInstanceServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Transactional
+public class StudyInstanceServiceImpl implements StudyInstanceService {
+
+	public StudyInstanceServiceImpl() {
+		// no-arg constuctor is required by CGLIB proxying used by Spring 3x and older.
+	}
+
+	public StudyInstanceServiceImpl(final HibernateSessionProvider sessionProvider) {
+		this.sessionProvider = sessionProvider;
+		this.daoFactory = new DaoFactory(sessionProvider);
+		this.experimentModelSaver = new ExperimentModelSaver(sessionProvider);
+	}
+
+	private HibernateSessionProvider sessionProvider;
+	private ExperimentModelSaver experimentModelSaver;
+	private DaoFactory daoFactory;
+
+	@Override
+	public StudyInstance createStudyInstance(final CropType crop, final int datasetId, final int instanceNumber) {
+
+		// Get the existing environment dataset variables.
+		// Since we are creating a new study instance, the values of these variables are just blank.
+		final List<MeasurementVariable> measurementVariables = this.daoFactory.getDmsProjectDAO().getObservationSetVariables(datasetId,
+			Arrays.asList(VariableType.ENVIRONMENT_DETAIL.getId(), VariableType.STUDY_CONDITION.getId()));
+
+		// The default value of an instance's location name is "Unspecified Location"
+		final Optional<Location> location = this.getUnspecifiedLocation();
+		final Geolocation geolocation =
+			this.createGeolocation(measurementVariables, instanceNumber, location.isPresent() ? location.get().getLocid() : null);
+		this.daoFactory.getGeolocationDao().save(geolocation);
+
+		final ExperimentValues experimentValue = new ExperimentValues();
+		experimentValue.setLocationId(geolocation.getLocationId());
+		final ExperimentModel experimentModel =
+			this.experimentModelSaver.addExperiment(crop, datasetId, ExperimentType.TRIAL_ENVIRONMENT, experimentValue);
+
+		final StudyInstance studyInstance = new StudyInstance();
+		studyInstance.setInstanceDbId(geolocation.getLocationId());
+		studyInstance.setInstanceNumber(instanceNumber);
+		studyInstance.setExperimentId(experimentModel.getNdExperimentId());
+		if (location.isPresent()) {
+			studyInstance.setLocationId(location.get().getLocid());
+			studyInstance.setLocationName(location.get().getLname());
+			studyInstance.setLocationAbbreviation(location.get().getLabbr());
+		}
+
+		return studyInstance;
+	}
+
+	@Override
+	public List<StudyInstance> getStudyInstances(final int studyId) {
+		final int environmentDatasetId =
+			this.daoFactory.getDmsProjectDAO().getDatasetsByTypeForStudy(studyId, DatasetTypeEnum.SUMMARY_DATA.getId()).get(0)
+				.getProjectId();
+		return this.daoFactory.getDmsProjectDAO().getDatasetInstances(environmentDatasetId);
+	}
+
+	protected Geolocation createGeolocation(final List<MeasurementVariable> measurementVariables, final int instanceNumber,
```

Comment:
We can probably extract to new class _GeolocationGenerator_ and use in ExperimentModelGenerator (replace _createNewGeoLocation_() method).

We want to be able to reuse these generator classes and new services to eventually replace the `WorkbookSaver` (and other related saver classes).

Reply:
Done.